src/libraries/trading/PlaceOrder.sol

```diff
@@ -0,0 +1,135 @@
+// Copyright (C) 2015 Forecast Foundation OU, full GPL notice in LICENSE
+
+// Bid / Ask actions: puts orders on the book
+// price is always in 10**18 fxp, so a price of 1 is 1 WEI.
+// amount is the number of attoshares the order is for (either to buy or to sell). For a currency with 18 decimals [like ether] if you buy 10**18 at a price of 10**18 then that's going to buy you ONE share [10**18 units] at a cost of ONE ETH [10**18 wei]. For a currency with say 9 decimals, if you buy 10**9 at a price of 10**18 that'll also buy you ONE full unit of that currency worth of shares. If you buy 10**9 at a price of 10**17 that'll buy you POINT_ONE full units of that currency worth of shares [so it'll cost you 10**8]. If you buy 10**8 amount at a price of 10**18 you're also effectively paying POINT_ONE units of currency, this time it's just to get you 10x less shares [in other words you're paying 10x more per share].
+// price is the exact price you want to buy/sell at [which may not be the cost, for example to short a binary market it'll cost 1-price, to go long it'll cost price]
+// smallest order value is 10**14 WEI
+
+pragma solidity ^0.4.13;
+
+import 'ROOT/IController.sol';
+import 'ROOT/libraries/math/SafeMathUint256.sol';
+import 'ROOT/trading/Trading.sol';
+import 'ROOT/trading/IOrders.sol';
+
+
+library PlaceOrder {
```

Comment:
I wonder if we can come up with a better name for this.  Fundamentally, it should be named like we would name a class that holds the items in `data`.  I think my suggestion would be to name it `Order`.  I'm a mildly worried that may conflict with other objects named Order (like in my taking branch) leading to confusion, though I think in this case it is the most appropriate name and perhaps I can integrate it into my branch when the time comes.

Reply:
Yeah, I considered naming it `Order` as well, but I was also concerned about potential naming conflicts.  If you don't think it's likely to pose too big of a problem, I'll go ahead and rename it to `Order`.